autoprecompiles/src/bitwise_lookup_optimizer.rs

```diff
@@ -108,7 +108,9 @@ pub fn optimize_bitwise_lookup<T: FieldElement, V: Hash + Eq + Clone + Ord + Deb
             multiplicity: One::one(),
         });
     }
-    system.algebraic_constraints.extend(new_constraints);
+    system
+        .algebraic_constraints
+        .extend(new_constraints.into_iter().filter(|expr| !expr.is_zero()));
```

Comment:
Not blocking, but imo ideally `system` or `algebraic_constraints` would take care of filtering out trivial constraints like this, not the caller like here.

Reply:
I know, but it's difficult because we need to re-index whenever we remove a constraint (because the references are index based) and I prefer to do that in bulk and not for each `substitute` call. But yeah, maybe we could replace that by a `compact()` call or something on the constraint system.